tools/reckless

```diff
@@ -0,0 +1,684 @@
+    test_log = []
+    with test.stderr:
+        for line in test.stderr:
+            test_log.append(line.strip('\n'))
+    test.wait()
+    # FIXME: add noexec test/warning. Maybe try chmod entrypoint.
+    if test.returncode != 0:
+        verbose("plugin testing error:")
+        for line in test_log:
+            verbose(f'  {line}')
+        print('plugin testing failed')
+        return False
+
+    # Find this cute little plugin a forever home
+    shutil.copytree(plugin_path, inst_path)
+    print(f'plugin installed: {inst_path}')
+    remove_dir(clone_path)
+    return True
+
+
+def install(plugin_name):
+    """reckless install <plugin>
+    downloads plugin from source repos, installs and activates plugin"""
+    assert isinstance(plugin_name, list)
+    plugin_name = plugin_name[0]
+    if plugin_name is None:
+        print('missing argument: plugin_name')
+
+    src = search([plugin_name])
+
+    if src:
+        verbose(f'Retrieving {plugin_name} from {src.repo}')
+        if not _install_plugin(src):
+            print('installation aborted')
+            sys.exit(1)
+        inst_path = os.path.join(RECKLESS_CONFIG.reckless_dir,
+                                 src.name,
+                                 src.entry)
+        RECKLESS_CONFIG.enable_plugin(inst_path)
+        enable([plugin_name])
+
+
+def uninstall(plugin_name):
+    """reckless uninstall <plugin>
+    disables plugin and deletes the plugin's reckless dir"""
+    assert isinstance(plugin_name, list)
+    plugin_name = plugin_name[0]
+    if plugin_name is not None:
+        # FIXME: Do something here.
+        print('Uninstalling plugin {}'.format(plugin_name))
+        disable([plugin_name])
+        plugin_dir = os.path.join(RECKLESS_CONFIG.reckless_dir, plugin_name)
+        verbose("looking for {}".format(plugin_dir))
+        if remove_dir(plugin_dir):
+            print(f"{plugin_name} uninstalled successfully.")
+
+
+def search(plugin_name):
+    """reckless search <plugin>
+    searches plugin index for plugin"""
+    plugin_name = plugin_name[0]
+    if plugin_name is None:
+        print('plugin name required')
+        return None
+    ordered_repos = RECKLESS_SOURCES
+    for r in RECKLESS_SOURCES:
+        if r.split('/')[-1].lower() == plugin_name.lower():
+            ordered_repos.remove(r)
+            ordered_repos.insert(0, r)
+    for r in ordered_repos:
+        p = _search_repo(plugin_name, r)
+        if p:
+            print(f"found {p.name} in repo: {p.repo}")
+            verbose(f"entry: {p.entry}")
+            if p.subdir:
+                verbose(f'sub-directory: {p.subdir}')
+            return p
+    print(f'Unable to locate source for plugin {plugin_name}')
+
+
+def lightning_cli_available():
+    clncli = Popen(['lightning-cli'], stdout=PIPE, stderr=PIPE)
+    clncli.wait(timeout=1)
+    if clncli.returncode == 0:
+        return True
+    else:
+        return False
```

Comment:
We could also check if we have an executable on the `$PATH`:

```
from distutils.spawn import find_executable
find_executable('lightning-cli') 
```

Reply:
Wow, this is so much nicer!